### PR TITLE
[OSF-7061] Use Browserchannel (long polling) over Websockets for sharejs

### PR DESCRIPTION
**Pull request by hmoco**
## Purpose
Remove websockets as it is not compatible with Google's load balancing.

## Changes
Switch use of websockets for the live editing feature present in file and wiki editing.

## Side effects
Potentially slower. Needs testing.

## Ticket
https://openscience.atlassian.net/browse/OSF-7061

### CODE REVIEW

File: website/addons/wiki/shareServer.js [hunk 1]
```diff
@@ -60,31 +60,16 @@ app.use(morgan('common'));
 
 // Allow CORS
 app.use(function(req, res, next) {
-    res.header('Access-Control-Allow-Origin', settings.corsAllowOrigin);
+    res.header('Access-Control-Allow-Origin', 'http://localhost:5000');
```
Comment (icereval) [inline website/addons/wiki/shareServer.js]: This needs to be configurable for non-local environments, also https://github.com/CenterForOpenScience/osf.io/pull/6681/files#diff-1ccc7cbdcef56c872e6db4c76737d8fcR19 indicates it is defaulted to this value.

File: website/addons/wiki/shareServer.js [hunk 2]
```diff
@@ -60,31 +60,16 @@ app.use(morgan('common'));
...
     res.header('Access-Control-Allow-Headers', 'Origin, X-Requested-With, Content-Type, Accept');
+    res.header('Access-Control-Allow-Credentials', true);
```
Comment (icereval) [inline website/addons/wiki/shareServer.js]: I think this line can be removed, you should not need to send the cookie credentials.

File: website/addons/wiki/shareServer.js [hunk 3]
```diff
@@ -140,7 +118,7 @@ wss.on('connection', function(client) {
 
             // Attach metadata to the client object
             client.userMeta = data;
-            wss.broadcast(docId, JSON.stringify({type: 'meta', users: docs[docId]}));
+            client.send( JSON.stringify({type: 'meta', users: docs[docId]}));
```
Comment (icereval) [inline website/addons/wiki/shareServer.js]: spacing

File: website/addons/wiki/shareServer.js [hunk 4]
```diff
@@ -177,15 +155,14 @@ wss.on('connection', function(client) {
                 }
             }
 
-            wss.broadcast(docId, JSON.stringify({type: 'meta', users: docs[docId]}));
+            client.send( JSON.stringify({type: 'meta', users: docs[docId]}));
```
Comment (icereval) [inline website/addons/wiki/shareServer.js]: spacing

File: website/addons/wiki/shareServer.js [hunk 5]
```diff
@@ -195,11 +172,11 @@ wss.on('connection', function(client) {
 
     // Give the stream to sharejs
     return share.listen(stream);
-});
+}));
 
 // Update a document from storage
 app.post('/reload/:id', jsonParser, function (req, res, next) {
-    wss.broadcast(req.params.id, JSON.stringify({
+    client.send( JSON.stringify({
```
Comment (icereval) [inline website/addons/wiki/shareServer.js]: spacing

File: website/addons/wiki/shareServer.js [hunk 6]
```diff
@@ -210,15 +187,15 @@ app.post('/reload/:id', jsonParser, function (req, res, next) {
 // Lock a document
 app.post('/lock/:id', function (req, res, next) {
     locked[req.params.id] = true;
-    wss.broadcast(req.params.id, JSON.stringify({type: 'lock'}));
+    client.send( JSON.stringify({type: 'lock'}));
```
Comment (icereval) [inline website/addons/wiki/shareServer.js]: spacing

File: website/addons/wiki/shareServer.js [hunk 7]
```diff
@@ -210,15 +187,15 @@ app.post('/reload/:id', jsonParser, function (req, res, next) {
...
     console.info('[Document Locked] docId: %s', req.params.id);
     res.send(util.format('%s was locked.', req.params.id));
 });
 
 // Unlock a document
 app.post('/unlock/:id', jsonParser, function (req, res, next) {
     delete locked[req.params.id];
-    wss.broadcast(req.params.id, JSON.stringify({
+    client.send( JSON.stringify({
```
Comment (icereval) [inline website/addons/wiki/shareServer.js]: spacing

File: website/addons/wiki/shareServer.js [hunk 8]
```diff
@@ -228,7 +205,7 @@ app.post('/unlock/:id', jsonParser, function (req, res, next) {
 
 // Redirect from a document
 app.post('/redirect/:id/:redirect', function (req, res, next) {
-    wss.broadcast(req.params.id, JSON.stringify({
+    client.send( JSON.stringify({
```
Comment (icereval) [inline website/addons/wiki/shareServer.js]: spacing

File: website/addons/wiki/shareServer.js [hunk 9]
```diff
@@ -238,7 +215,7 @@ app.post('/redirect/:id/:redirect', function (req, res, next) {
 
 // Redirect from a deleted document
 app.post('/delete/:id/:redirect', function (req, res, next) {
-    wss.broadcast(req.params.id, JSON.stringify({
+    client.send( JSON.stringify({
```
Comment (icereval) [inline website/addons/wiki/shareServer.js]: spacing

File: website/addons/wiki/shareServer.js [hunk 10]
```diff
@@ -45,7 +45,7 @@ var share = sharejs.server.createClient({backend: backend});
 var app = express();
 var jsonParser = bodyParser.json();
 var server = http.createServer(app);
-var wss = new WebSocketServer({server: server});
+
```
Comment (icereval) [inline website/addons/wiki/shareServer.js]: spacing

**Comment by hmoco** [comment]
@icereval once tests pass, this is good for another look

**Comment by brianjgeiger** [comment]
@hmoco Could you re-merge and make sure tests still pass?

**Pull request closed without merging**